src/lib/registration/clients/scc_auto.rb

```diff
@@ -198,7 +197,16 @@ def online_medium_config
         Yast::Packages.ImportGPGKeys
 
         products = Y2Packager::ProductControlProduct.products
-        ay_product = Yast::Profile.current.fetch("software", {}).fetch("products", []).first
+
+        Yast.import "AutoinstFunctions"
+
+        selected_product = Yast::AutoinstFunctions.selected_product
+        log.info "selected product #{selected_product.inspect}"
+        ay_product = if selected_product.respond_to?(:name)
```

Comment:
So `Yast::AutoinstFunctions.selected_product` can return different things? According to the documentation, it is expected to return a [Y2Packager::Product](https://github.com/yast/yast-autoinstallation/blob/58f204a4ad7e47b59e892e3bb2fb9d4cb7d71f30/src/modules/AutoinstFunctions.rb#L84) instance. Perhaps I am missing something.

Reply:
sadly it now can return both due to new medium layout where it can in some phases return controlProduct. I will adapt documentation. See https://github.com/yast/yast-autoinstallation/blob/58f204a4ad7e47b59e892e3bb2fb9d4cb7d71f30/src/modules/AutoinstFunctions.rb#L150